File: app/src/main/java/it/amonshore/comikkua/ui/releases/IReleaseViewHolderCallback.java
package it.amonshore.comikkua.ui.releases;

/**
 * Callback per i ViewHoler riferiti alle release.
 */
interface IReleaseViewHolderCallback {

    /**
     * Notifica il click sull'intera release.
     *
     * @param comicsId id del comics
     * @param id       id della release
     * @param position posizione del Viewholer
     */
    void onReleaseClick(long comicsId, long id, int position);

    /**
     * Notifica la selezione del pulsante menu.
     *
     * @param comicsId id del comics
     * @param id       id della release
     * @param position posizione del Viewholer
     */
    void onReleaseMenuSelected(long comicsId, long id, int position);
}

File: app/src/main/java/it/amonshore/comikkua/ui/releases/IReleaseViewHolderCallback.kt
package it.amonshore.comikkua.ui.releases

import it.amonshore.comikkua.data.release.IReleaseViewModelItem

interface IReleaseViewHolderCallback {
    fun onReleaseClick(item: IReleaseViewModelItem, position: Int)
    fun onReleaseMenuSelected(item: IReleaseViewModelItem, position: Int)
}